src/servers/grpc_server.cc

```diff
@@ -151,9 +156,25 @@ AsyncResources::ResponseAlloc(
   // that the number and order of raw output entries equals the output
   // meta-data.
   std::string* raw_output = response->add_raw_output();
-  if (byte_size > 0) {
```

Comment:
When byte_size is 0 you shouldn't do anything with shared memory. Keep the (byte_size > 0) as condition around all logic.

Reply:
Fixed